### PR TITLE
Replace MAX_PATH with new defines, rest of coreclr.

### PR DESCRIPTION
**Pull request by Priya91**
@jkotas @JeremyKuhne

### CODE REVIEW

File: src/palrt/path.cpp [hunk 1]
```diff
@@ -280,7 +280,7 @@ STDAPI_(BOOL) PathCanonicalizeW(LPWSTR lpszDst, LPCWSTR lpszSrc)
     BOOL fUNC;
     int cchPC;
 
-    RIPMSG(lpszDst && IS_VALID_WRITE_BUFFER(lpszDst, WCHAR, MAX_PATH), "PathCanonicalize: caller passed bad lpszDst");
+    RIPMSG(lpszDst && IS_VALID_WRITE_BUFFER(lpszDst, WCHAR, MAX_LONGPATH), "PathCanonicalize: caller passed bad lpszDst");
```
Comment (jkotas) [inline src/palrt/path.cpp]: The spec for PathCanonicalize is hardcoded to use MAX_PATH - https://msdn.microsoft.com/en-us/library/windows/desktop/bb773569(v=vs.85).aspx . We should not be changing to use MAX_LONGPATH for PAL.

**Comment by Priya91** [comment]
@jkotas Can you please review for signoff.. Thanks!

File: src/palrt/path.cpp [hunk 2]
```diff
@@ -466,7 +466,7 @@ STDAPI_(LPWSTR) PathCombineW(LPWSTR lpszDest, LPCWSTR lpszDir, LPCWSTR lpszFile)
                 {
                     // Skip the backslash when copying
                     // Note: We don't support strings longer than 4GB, but that's
-                    // okay because we already barf at MAX_PATH
+                    // okay because we already barf at MAX_LONGPATH
```
Comment (jkotas) [inline src/palrt/path.cpp]: This one should one stay at MAX_PATH

File: src/palrt/path.cpp [hunk 3]
```diff
@@ -541,7 +541,7 @@ STDAPI_(LPWSTR) PathAddBackslashW(LPWSTR lpszPath)
                     // if we find these cases, return NULL.  Note: We need to
                     // check those places that call us to handle their GP fault
                     // if they try to use the NULL!
-                    if (ichPath >= (MAX_PATH - 2)) // -2 because ichPath doesn't include NULL, and we're adding a CH_WHACK.
+                    if (ichPath >= (MAX_LONGPATH - 2)) // -2 because ichPath doesn't include NULL, and we're adding a CH_WHACK.
```
Comment (jkotas) [inline src/palrt/path.cpp]: This one should also stay at MAX_PATH

**Comment by jkotas** [comment]
LGTM otherwise.

**Comment by Priya91** [comment]
@dotnet-bot test this please

**Pull request closed without merging**